evo-delivery-app/evo-delivery-backend/src/clients/EaClientImpl.ts

```diff
@@ -0,0 +1,111 @@
+import { EA_ENGINE_API_V1_URL } from '../configs';
+import {
+  DistanceMatrix,
+  Driver,
+  EaComponentDetails,
+  EaComponentTypes,
+  EaEvaluateArgs,
+  EaEvaluateHttpRequestArgs,
+  EaEvaluateHttpRequestKwargs,
+  EaEvaluateHttpResponse,
+  EaEvaluateKwargs,
+  EaHttpRequestDistances,
+  EaHttpRequestDriver,
+  EaHttpRequestOrder,
+  Order,
+} from '../types';
+import { convertObjCamelToSnakeCase } from '../utils';
+import { EaClient } from './EaClient';
+import { HttpClient } from './HttpClient';
+
+export class EaClientImpl implements EaClient {
+  private readonly httpClient: HttpClient;
+  private readonly baseUrl: string;
+  private readonly apiUrlPrefix: string = EA_ENGINE_API_V1_URL;
+
+  constructor(httpClient: HttpClient, baseUrl: string) {
+    this.httpClient = httpClient;
+    this.baseUrl = baseUrl;
+  }
+
+  private convertToEaHttpDriver({ id, maxCapacity, maxDistance }: Driver) {
+    return convertObjCamelToSnakeCase({
+      id,
+      maxCapacity,
+      maxDistance,
+    }) as EaHttpRequestDriver;
+  }
+
+  private convertToEaHttpOrder({ id, weight }: Order) {
+    return convertObjCamelToSnakeCase({
+      id,
+      weight,
+    }) as EaHttpRequestOrder;
+  }
+
+  private convertToEaHttpDistances(distanceMatrix: DistanceMatrix) {
+    return Object.entries(distanceMatrix).reduce(
+      (distancesAccumulator, [from, rowData]) => {
+        distancesAccumulator[from] = Object.entries(rowData).reduce(
+          (rowDataAccumulator, [to, colData]) => {
+            rowDataAccumulator[to] = colData.distance.value;
+            return rowDataAccumulator;
```

Comment:
lol, I wish I could understand what's going on, but it's looking good... will write test for this when we will have time